scripts/release/mule/README.md

```diff
@@ -0,0 +1,216 @@
+
+- `ARCH_TYPE`, i.e., `amd64`
+- `BRANCH`
+- `CHANNEL`
+- `OS_TYPE`
+- `VERSION`
+
+# Build Stages
+
+- [package](#package)
+- [upload](#upload)
+- [test](#test)
+- [sign](#sign)
+- [deploy](#deploy)
+
+## package
+
+- see `./go-algorand/package.yaml`
+
+#### `mule` jobs
+
+    - package
+        + packages `deb`, `rpm` and `docker`
+
+    - package-deb
+        + packages only `deb`
+
+    - package-rpm:
+        + packages only `rpm`
+
+    - package-docker
+        + packages docker image
+
+## upload
+
+- see `./go-algorand/package-test.yaml`
+
+- customizable environment variables:
+
+    + `CHANNEL`
+    + `VERSION`
+
+#### `mule` jobs
+
+    - package-upload
+
+## test
+
+- see `./go-algorand/package-test.yaml`
+
+- customizable environment variables:
+
+    + `ARCH_BIT`, i.e., the value from `uname -m`
+    + `NETWORK`
+    + `SHA`, i.e., the value from `git rev-parse HEAD` if not passed on CLI
+    + `USE_CACHE`
+
+#### `mule` jobs
+
+    - package-test
+        + tests both `deb` and `rpm`
+
+    - package-test-deb
+        + tests only `deb`
+
+    - package-test-rpm
+        + tests only `rpm`
+
+## sign
+
+- see `./go-algorand/package-sign.yaml`
+
+- customizable environment variables:
+
+    + `ARCH_BIT`, i.e., the value from `uname -m`
+    + `USE_CACHE`
+
+### `mule` jobs
+
+    - package-sign
+        + signs both `deb` and `rpm`
+
+## deploy
+
+- see `./go-algorand/package-deploy.yaml`
+
+- customizable environment variables:
+
+    + `NETWORK`
+    + `NO_DEPLOY`
+    + `PACKAGES_DIR`
+
+#### `mule` jobs
+
+    - package-deploy
+        + deploys both `deb` and `rpm`
+
+    - package-deploy-deb
+        + deploys only `deb`
+
+    - package-deploy-rpm
+        + deploys only `rpm`
+
+# Custom Builds
+
+It is sometimes necessary to create packages after doing a local build.
+
+For example, the packaging build process will look like this:
+
+```
+mule -f package.yaml package
+```
+
+This can produce packages like the following:
+
+```
+algorand_dev_linux-amd64_2.1.86615.deb
+algorand-devtools_dev_linux-amd64_2.1.86615.deb
+```
+
+Note that this is in the format `{ALGORAND_PACKAGE_NAME}_{CHANNEL}_{OS_TYPE}-{ARCH_TYPE}_{VERSION}.deb. `rpm` packages will follow their own format which is easy to intuit.
+
+It is common that a custom build is performed on a feature branch other than `rel/stable` or `rel/beta` and that the build environment will need to be modified. In these instances, it is important to be able to pass values to the build process to customize a build.
+
+The most common way to do this is to modify the environment that the subprocess inherits by specifying the values on the command *before* the command.  This won't need to be done for the package stage, but often needs to be done with subsequent stages.
+
+In order to be able to correctly run some of the stages, such as testing and signing, several values needed by the subsequent stages must be explicitly passed to those stages.
```

Comment:
several values are needed*

(I'm not sure how to do the suggest changes thing that John does)